tests/unit/multiSig.js

```diff
@@ -0,0 +1,134 @@
+            await expect(
+                multiSigSecp.getPubK()
+            ).to.eventually.be.fulfilled;
+        });
+
+        it('Success: Get Address when a Public Key exists', async () => {
+            await multiSigSecp.addPublicKeys(publicKeys)
+            await expect(
+                multiSigSecp.getAddress()
+            ).to.eventually.be.fulfilled;
+        });
+    });
+
+    describe('Sign and Multi Sign', () => {
+        it('Fail: Reject Sign when called with invalid rawMsg', async () => {
+            await expect(
+                multiSigSecp.sign(null)
+            ).to.eventually.be.rejectedWith('Missing input');
+        });
+            
+        it('Fail: Reject Sign Multi when called with invalid rawMsgs', async () => {
+            await expect(
+                multiSigSecp.signMulti(null)
+            ).to.eventually.be.rejectedWith(Error);
+        });
+
+        it('Success: Sign one message', async () => {
+            await multiSigSecp.addPublicKeys(publicKeys)
+            await expect(
+                multiSigSecp.sign('0000ffeebabe')
+            ).to.eventually.be.fulfilled;
+        });
+
+        it('Success: Sign Multi messages', async () => {
+            await multiSigSecp.addPublicKeys(publicKeys)
+            await expect(
+                multiSigSecp.signMulti(['0000ffeebabe', '0000ffeebabe'])
+            ).to.eventually.be.fulfilled;
+        });
+    });
+
+    describe('Aggregate and Verify Signatures', () => {
+        it('Fail: Reject Aggregate Signatures when called with invalid argument', async () => {
+            await expect(
+                multiSigSecp.aggregateSignatures(null)
+            ).to.eventually.be.rejectedWith(Error);
+        });
+
+        it('Fail: Reject Aggregate Multi Signatures when called with invalid argument', async () => {
+            await expect(
+                multiSigSecp.aggregateSignaturesMulti(null)
+            ).to.eventually.be.rejectedWith(Error);
+        });
+
+        it('Fail: Reject Verify Aggregate when called with invalid argument', async () => {
+            await expect(
+                multiSigSecp.verifyAggregate(null)
+            ).to.eventually.be.rejectedWith(Error);
+        });
+
+        it('Fail: Reject Verify Aggregate when called with invalid sig hex length', async () => {
+            await expect(
+                multiSigSecp.verifyAggregate(msgHex, '0xc0ffeebab')
+            ).to.eventually.be.rejectedWith(Error);
+        });
+
+        it('Success: Aggregate Signatures', async () => {
+            await expect(
+                multiSigSecp.aggregateSignatures(signatures)
+            ).to.eventually.be.fulfilled;
+        });
+
+        it('Success: Aggregate Multi Signatures', async () => {
+            await expect(
+                multiSigSecp.aggregateSignaturesMulti([signatures])
+            ).to.eventually.be.fulfilled;
+        });
+    });
+});
```

Comment:
aggregateSignatures should probably fail unless an array is provided and these test cases should reflect that